org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/ui/trace/TraceForTypeRootProvider.java

```diff
@@ -73,6 +76,10 @@
 	@Inject
 	private ITraceForStorageProvider traceForStorageProvider;
 	
+	@Inject(optional=true) 
```

Comment:
i had a look. this is not needed. we assume the workspace beeing there all over the place. => you can remove the optional=true and nullable